src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,159 @@
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validateRangeDoubleParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validatePositiveIntegerParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validateStringParameters;
+
+/**
+ * The implementation {@link Chunker} for fixed token length algorithm.
+ */
+public class FixedTokenLengthChunker implements Chunker {
+
+    public static final String ALGORITHM_NAME = "fixed_token_length";
+
+    // field name for each parameter
+    public static final String ANALYSIS_REGISTRY_FIELD = "analysis_registry";
+    public static final String TOKEN_LIMIT_FIELD = "token_limit";
+    public static final String OVERLAP_RATE_FIELD = "overlap_rate";
+    public static final String MAX_TOKEN_COUNT_FIELD = "max_token_count";
+    public static final String TOKENIZER_FIELD = "tokenizer";
+
+    // default values for each parameter
+    private static final int DEFAULT_TOKEN_LIMIT = 384;
+    private static final double DEFAULT_OVERLAP_RATE = 0.0;
+    private static final int DEFAULT_MAX_TOKEN_COUNT = 10000;
+    private static final String DEFAULT_TOKENIZER = "standard";
+
+    // parameter restrictions
+    private static final double OVERLAP_RATE_LOWER_BOUND = 0.0;
+    private static final double OVERLAP_RATE_UPPER_BOUND = 0.5;
+    private static final Set<String> WORD_TOKENIZERS = Set.of(
+        "standard",
+        "letter",
+        "lowercase",
+        "whitespace",
+        "uax_url_email",
+        "classic",
+        "thai"
+    );
+
+    // parameter value
+    private int tokenLimit;
+    private String tokenizer;
+    private double overlapRate;
+    private final AnalysisRegistry analysisRegistry;
+
+    public FixedTokenLengthChunker(Map<String, Object> parameters) {
+        validateAndParseParameters(parameters);
+        this.analysisRegistry = (AnalysisRegistry) parameters.get(ANALYSIS_REGISTRY_FIELD);
+    }
+
+    /**
+     * Validate and parse the parameters for fixed token length algorithm,
+     * will throw IllegalArgumentException when parameters are invalid
+     *
+     * @param parameters a map containing parameters, containing the following parameters:
+     * 1. tokenizer: the <a href="https://opensearch.org/docs/latest/analyzers/tokenizers/index/">analyzer tokenizer</a> in opensearch
+     * 2. token_limit: the token limit for each chunked passage
+     * 3. overlap_rate: the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     * 4. max_token_count: the max token limit for the tokenizer
+     * Here are requirements for parameters:
+     * max_token_count and token_limit should be a positive integer
+     * overlap_rate should be within range [0, 0.5]
+     * tokenizer should be string
+     */
+    @Override
+    public void validateAndParseParameters(Map<String, Object> parameters) {
```

Comment:
method should clearly say it set or initialize variables, basically mutates the state. One more point to split validation and parser logic

Reply:
I will split the function into two parts: validate and parse.

Reply:
Latest update: merge `validate` into `parse`